File: src/Entity/Bid.php
<?php

namespace App\Entity;

use App\Repository\BidRepository;
use DateTime;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: BidRepository::class)]
class Bid
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column]
    private ?float $bidAmount = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    private ?\DateTimeInterface $entryDate = null;

    #[ORM\Column(type: Types::TIME_MUTABLE)]
    private ?\DateTimeInterface $entryTime = null;

    #[ORM\Column(length: 255)]
    private ?string $type = null;

    #[ORM\ManyToOne(inversedBy: 'bids')]
    #[ORM\JoinColumn(nullable: false)]
    private ?User $user = null;

    #[ORM\ManyToOne(inversedBy: 'bids')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Produit $produit = null;
    
    public function __construct()
    {
        $this->entryDate = new \DateTime("today");
        $this->entryTime = new \DateTime("now");
        $this->type = "Active";
    }
    public function __withID(?int $prodid)
    {
        $this->entryDate = new \DateTime("today");
        $this->entryTime = new \DateTime("now");
        $this->type = "Active";
        $this->produit = $prodid;
        return $this;
    }
    
    public static function create() {
        return new self();
    }
    
    public function getId(): ?int
    {
        return $this->id;
    }

    public function getBidAmount(): ?float
    {
        return $this->bidAmount;
    }

    public function setBidAmount(float $bidAmount): self
    {
        $this->bidAmount = $bidAmount;

        return $this;
    }

    public function getEntryDate(): ?\DateTimeInterface
    {
        return $this->entryDate;
    }

    public function setEntryDate(\DateTimeInterface $entryDate): self
    {
        $this->entryDate = $entryDate;

        return $this;
    }

    public function getEntryTime(): ?\DateTimeInterface
    {
        return $this->entryTime;
    }

    public function setEntryTime(\DateTimeInterface $entryTime): self
    {
        $this->entryTime = $entryTime;

        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): self
    {
        $this->type = $type;

        return $this;
    }

    public function getUser(): ?User
    {
        return $this->user;
    }

    public function setUser(?User $user): self
    {
        $this->user = $user;

        return $this;
    }

    public function getProduit(): ?Produit
    {
        return $this->produit;
    }

    public function setProduit(?Produit $produit): self
    {
        $this->produit = $produit;

        return $this;
    }
}

File: src/entite/Bid.java

package entite;
import java.sql.Date;
import java.sql.Time;


public class Bid  {
    private int id_bid;
    private int id_user;
    private int id_listing;
    private int bid_amount;
    private Date entry_date;
    private Time entry_time;
    private int active_duration;
    private String type;


    public Bid() {
    }

    public Bid(int id_user, int id_listing, int bid_amount, String type) {
        this.id_user = id_user;
        this.id_listing = id_listing;
        this.bid_amount = bid_amount;
        this.type = type;
    }
    public Bid(int bid_amount, String type) {
        this.bid_amount = bid_amount;
        this.type = type;
    }
    

    public int getId_bid() {
        return id_bid;
    }

    public int getBid_amount() {
        return bid_amount;
    }

    public String getType() {
        return type;
    }

    public void setId_bid(int id_bid) {
        this.id_bid = id_bid;
    }

    public void setBid_amount(int bid_amount) {
        this.bid_amount = bid_amount;
    }
    
    public Date getEntry_date() {
        return entry_date;
    }

    public void setEntry_date(Date entry_date) {
        this.entry_date = entry_date;
    }

    public Time getEntry_time() {
        return entry_time;
    }

    public void setEntry_time(Time entry_time) {
        this.entry_time = entry_time;
    }

    public void setType(String type) {
        this.type = type;
    }

    public int getId_user() {
        return id_user;
    }

    public void setId_user(int id_user) {
        this.id_user = id_user;
    }

    public int getId_listing() {
        return id_listing;
    }

    public void setId_listing(int id_listing) {
        this.id_listing = id_listing;
    }

    public int getActive_duration() {
        return active_duration;
    }

    public void setActive_duration(int active_duration) {
        this.active_duration = active_duration;
    }

    @Override
    public String toString() {
        return "Bid{" +
                "id_bid=" + id_bid +
                ", id_user=" + id_user +
                ", id_listing=" + id_listing +
                ", bid_amount=" + bid_amount +
                ", entry_date=" + entry_date +
                ", entry_time=" + entry_time +
                ", type='" + type + '\'' +
                '}';
    }


}
